### DISCUSSION TITLE
FMU-proxy to run FMUs in Docker containers

### DISCUSSION THREAD

**Discussion by Ju-Ru** [NTNU-IHB/FMU-proxy#132] [Q&A] [answered]
Hello everyone,

I would be interested to understand how the solution can be used to run FMUs as independent Docker containers. Are there any examples I can start with (i.e. Dockerfiles)? My FMUs have Windows dependencies and, yes, I would run them in Windows containers. I did this before, but having them separated from the main application would be awesome. 

Also, how would I interact with the FMU from my C# .NET application in this case? Maybe I missed relevant information in the repository files - in this case, some pointers would be sufficient. 

Thanks in advance!

**Comment by markaren** [comment, marked as answer]
I don't use docker, but all you have to do is to copy paste the `fmu-proxy-booter` executable into the container.

The docker container also needs a JDK:
JAVA_HOME should point to a JDK8+ installation.
%JAVA_HOME%\jre\bin\server must be added to PATH.

Basically:
1. Clone the project
2. Run `./gradlew build`
3. Run `fmu-proxy-booter` (located in fmu-proxy-booter/build/install) in the container supplied with an open port.
4. Run `fmu-proxify` located under `fmu-proxify/build/install` on the target FMU. Supply the remoteAddress. I.e. IP of the container and port used by the booter. 
5. Load the proxified FMU in any FMI 2.0 compatible tool.

The README should cover this.

**Reply by Ju-Ru** [reply]
Thanks a lot! I will give it a try.
